codec-http/src/main/java/io/netty/handler/codec/http/Cookie.java

```diff
@@ -20,78 +20,76 @@
 /**
  * An interface defining an
  * <a href="http://en.wikipedia.org/wiki/HTTP_cookie">HTTP cookie</a>.
+ * @deprecated Use {@link io.netty.handler.codec.http.cookie.Cookie} instead.
  */
-public interface Cookie extends Comparable<Cookie> {
+@Deprecated
+public interface Cookie extends io.netty.handler.codec.http.cookie.Cookie {
 
     /**
-     * Returns the name of this {@link Cookie}.
-     *
-     * @return The name of this {@link Cookie}
+     * @deprecated Use {@link #name()} instead.
      */
+    @Deprecated
     String getName();
 
     /**
-     * Returns the value of this {@link Cookie}.
-     *
-     * @return The value of this {@link Cookie}
+     * @deprecated Use {@link #value()} instead.
      */
+    @Deprecated
     String getValue();
 
     /**
-     * Sets the value of this {@link Cookie}.
-     *
-     * @param value The value to set
-     */
-    void setValue(String value);
-
-    /**
-     * Returns the domain of this {@link Cookie}.
-     *
-     * @return The domain of this {@link Cookie}
+     * @deprecated Use {@link #domain()} instead.
      */
+    @Deprecated
     String getDomain();
 
     /**
-     * Sets the domain of this {@link Cookie}.
-     *
-     * @param domain The domain to use
-     */
-    void setDomain(String domain);
-
-    /**
-     * Returns the path of this {@link Cookie}.
-     *
-     * @return The {@link Cookie}'s path
+     * @deprecated Use {@link #path()} instead.
      */
+    @Deprecated
     String getPath();
 
     /**
-     * Sets the path of this {@link Cookie}.
-     *
-     * @param path The path to use for this {@link Cookie}
+     * @deprecated Use {@link #comment()} instead.
      */
-    void setPath(String path);
+    @Deprecated
+    String getComment();
 
     /**
      * Returns the comment of this {@link Cookie}.
      *
      * @return The comment of this {@link Cookie}
+     *
+     * @deprecated Not part of RFC6265
      */
-    String getComment();
+    @Deprecated
+    String comment();
 
     /**
      * Sets the comment of this {@link Cookie}.
      *
      * @param comment The comment to use
+     *
+     * @deprecated Not part of RFC6265
      */
+    @Deprecated
     void setComment(String comment);
 
     /**
+     * @deprecated Use {@link #maxAge()} instead.
+     */
+    @Deprecated
+    long getMaxAge();
+
+    /**
      * Returns the maximum age of this {@link Cookie} in seconds or {@link Long#MIN_VALUE} if unspecified
      *
      * @return The maximum age of this {@link Cookie}
+     *
+     * @deprecated Not part of RFC6265
      */
-    long getMaxAge();
+    @Deprecated
+    long maxAge();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1161)